### DISCUSSION TITLE
Stack Pull Requests for 1 dependency?

### DISCUSSION THREAD

**Discussion by janpio** [renovatebot/renovate#10561] [Request Help] [answered]
We have a (internal) dependency that when updated, often leads to snapshot changes in our tests that need to be reviewed and accepted before a PR can be merged (and fails until then). The dependency also gets updated quite frequently, the PRs are updated and rebased quite often.

Along the way we often get many _different_ snapshot changes, which as the PRs are not merged, just collect in the PR and becomes harder and harder to review over time as different changes with different snapshot changes are now part of 1 pull request.

Is there a way to avoid that somehow?
I though that it could be neat to have 1 PR per version of that dependency, that all stack onto each other. Version 1 leads to a failing test, so Version 2 gets a new PR. That way we can merge the first PR on its own, or merge the second one directly if that fixes a problem in the first one that was the source for the test failure. Does that make sense? Is that possible?

**Comment by rarkins** [comment, marked as answer]
If you mean have separate PRs per version then we don't support that today but in theory it could be done. Are you sure that it would be better than the current approach? You could end up with a lot of them. 

BTW Renovate should stop updating a PR if you've edited it so I'm not sure how that fits in with your description.

**Reply by janpio** [reply]
I am honestly not sure if it would be better, so would be willing to try it out if it already existed :D 

Most of our PRs just merge cleanly, but sometimes one comes along that makes the snapshots fail and then another one comes along to make it fail again, and then you have a PR that mixes the two different failures and you have to create a manual one for the earlier version again to cleanly be able to fix it.

With multiple PRs, after the first failure obviously each version would get an additional PR (so not sure that is actually a good idea...) but after fixing and merging the first one - in isolation with only its failures - the other ones would be rebased, potentially just pass tests and the newest passing one then can be merged, and the others closed.

The details... 🤷
